saleor/tests/e2e/checkout/test_checkout_products_on_fixed_promotion.py

```diff
@@ -0,0 +1,222 @@
+        e2e_staff_api_client,
+        promotion_id,
+        discount_type,
+        discount_value,
+        promotion_rule_name,
+        channel_id,
+        product_id,
+    )
+    product_predicate = promotion_rule["cataloguePredicate"]["productPredicate"]["ids"]
+    assert promotion_rule["channels"][0]["id"] == channel_id
+    assert product_predicate[0] == product_id
+
+    return product_variant_id
+
+
+@pytest.mark.e2e
+def test_checkout_products_on_fixed_promotion_core_2102(
+    e2e_not_logged_api_client,
+    e2e_staff_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_shipping,
+    permission_manage_product_types_and_attributes,
+    permission_manage_discounts,
+):
+    # Before
+    channel_slug = "test-channel"
+    variant_price = "20"
+    promotion_name = "Promotion Fixed"
+    discount_value = 5
+    discount_type = "FIXED"
+    promotion_rule_name = "rule for product"
+
+    product_variant_id = prepare_product(
+        e2e_staff_api_client,
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        permission_manage_discounts,
+        channel_slug,
+        variant_price,
+        promotion_name,
+        discount_value,
+        discount_type,
+        promotion_rule_name,
+    )
+
+    # Step 1 - checkoutCreate for product on promotion
+    lines = [
+        {"variantId": product_variant_id, "quantity": 2},
+    ]
+    checkout_data = checkout_create(
+        e2e_not_logged_api_client,
+        lines,
+        channel_slug,
+        email="testEmail@example.com",
+        set_default_billing_address=True,
+        set_default_shipping_address=True,
+    )
+    checkout_id = checkout_data["id"]
+    checkout_lines = checkout_data["lines"][0]
+    shipping_method_id = checkout_data["shippingMethods"][0]["id"]
+    unit_price = float(variant_price) - discount_value
+
+    assert checkout_data["isShippingRequired"] is True
+    assert checkout_lines["unitPrice"]["gross"]["amount"] == unit_price
+    assert checkout_lines["undiscountedUnitPrice"]["amount"] == float(variant_price)
+
+    # Step 2 - Set DeliveryMethod for checkout.
+    checkout_data = checkout_delivery_method_update(
+        e2e_not_logged_api_client,
+        checkout_id,
+        shipping_method_id,
+    )
+    assert checkout_data["deliveryMethod"]["id"] == shipping_method_id
+    total_gross_amount = checkout_data["totalPrice"]["gross"]["amount"]
+
+    # Step 3 - Create payment for checkout.
+    checkout_dummy_payment_create(
+        e2e_not_logged_api_client, checkout_id, total_gross_amount
+    )
+
+    # Step 5 - Complete checkout.
+    order_data = checkout_complete(e2e_not_logged_api_client, checkout_id)
+
+    order_line = order_data["lines"][0]
+    assert order_data["status"] == "UNFULFILLED"
+    assert order_data["total"]["gross"]["amount"] == total_gross_amount
```

Comment:
Could we also check subtotal?

Reply:
Added https://github.com/saleor/saleor/pull/13649/commits/2dbf9fb484ec55e7e5ff9588e57092c766b6b820